salina/agents/xformers_transformers.py

```diff
@@ -162,16 +227,48 @@ def forward(self, t: Optional[int] = None, **_):
 
             T = queries.size()[1]
 
-            attn_mask = self._get_mask(T, self.n_steps, tokens.device)  # n_steps x n_steps
+            if (
+                not _is_triton_available
```

Comment:
this is probably not smart enough: if triton is available and you're on gpu -> blocksparse. For real, probably that for small T then sparse is faster. Something else to note is that blocksparse will always be fp16 (even if you did not ask for it), because it uses the tensor cores and those are only available with fp16 on V100